services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala

```diff
@@ -39,61 +45,77 @@ class ProjectConfigurationServiceTest extends AsyncWordSpec with Matchers with M
+        when(configurationRepository.getConfigurationById(projectId))
+          .thenReturn(Future.successful(Some(activeConfiguration)))
         when(configurationRepository.updateConfiguration(activeConfiguration)).thenReturn(Future.failed(error))
-        configurationService.addConfiguration(activeConfiguration).failed.map(_ shouldBe error)
+        configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
+      }
+
+      "return failure creation if user is not owner for project" in {
+        val error = new ProjectAccessDeniedException
+        when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.failed(error))
+        configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
       }
     }
     "get configuration by project id" should {
       "return project if it was found" in {
         val result: Option[ProjectConfiguration] = Some(activeConfiguration)
-        when(configurationRepository.getById(projectId)).thenReturn(Future.successful(result))
-        configurationService.getById(projectId).map(_ shouldBe result)
+        when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future(dummyProject))
+        when(configurationRepository.getConfigurationById(projectId)).thenReturn(Future.successful(result))
+        configurationService.getConfigurationById(projectId, userId).map(_ shouldBe result)
       }
 
       "not return inactive project" in {
         val result: Option[ProjectConfiguration] = Some(inactiveConfiguration)
-        when(configurationRepository.getById(projectId)).thenReturn(Future.successful(result))
-        configurationService.getById(projectId).map(_ shouldBe None)
+        when(configurationRepository.getConfigurationById(projectId)).thenReturn(Future.successful(result))
+        configurationService.getConfigurationById(projectId, userId).map(_ shouldBe None)
       }
 
       "not fail if project wasn't found" in {
         val result: Option[ProjectConfiguration] = None
-        when(configurationRepository.getById(projectId)).thenReturn(Future.successful(result))
-        configurationService.getById(projectId).map(_ shouldBe result)
+        when(configurationRepository.getConfigurationById(projectId)).thenReturn(Future.successful(result))
+        configurationService.getConfigurationById(projectId, userId).map(_ shouldBe result)
       }
 
       "return failure if repository returned error" in {
         val error = new Exception("Oh no")
-        when(configurationRepository.getById(projectId)).thenReturn(Future.failed(error))
-        configurationService.getById(projectId).failed.map(_ shouldBe error)
+        when(configurationRepository.getConfigurationById(projectId)).thenReturn(Future.failed(error))
+        configurationService.getConfigurationById(projectId, userId).failed.map(_ shouldBe error)
+      }
+
+      "return failure if user is not owner for project" in {
```

Comment:
Maybe change wording a little to something like "return failure if user is not the project's owner"